lplatt_keep_window_static.R

```diff
@@ -0,0 +1,129 @@
+# static small window
+# model added to window to show it converging
+
+library(dplyr)
+
+plot_type <- switch(Sys.info()[['sysname']],
+                    Windows= "cairo",
+                    Linux  = "Xlib",
+                    Darwin = "quartz")
+
+time0 <- as.POSIXct("2019-11-01 07:30:00 UTC")
+time_min <- time0 - (2*3600*24)
+time_max <- time0 + (2*3600*24)
+
+##### Merge model info into one data frame #####
+mesonet_files <- list.files(path="1_fetch/out/", 
+                            pattern = "mesonet_data_(.*).rds$",
+                            full.names = TRUE)
+mesonet_all <- NULL
+for(file in mesonet_files) {
+  one_file <- readRDS(file)
+  mesonet_all <- bind_rows(mesonet_all, one_file)
+}
+
+##### Calculate how far in advance (in days) the forecast & format into columns #####
+
+models_in_window <- mesonet_all %>% 
+  mutate(issued = as.POSIXct(issued, "%Y-%m-%d %H:%M", tz="UTC")) %>% 
+  # Somehow, there are 31 "issued" dates that are greater than "valid"
+  filter(issued < valid) %>%
+  filter(valid >= time_min, valid <= time_max) %>%
+  mutate(forecast_range_days = as.numeric(difftime(issued, valid, units = "days")),
+         forecast_range_round = ceiling(forecast_range_days/0.5) * 0.5 * -1,
+         forecast_name = paste0("Pred_Stage_", sprintf("%.01f", forecast_range_round))) %>% 
+  select(valid, forecast_name, primary_value) %>%  
+  distinct() %>% 
+  # Due to rounding the forecast range, some entries have the exact same identifier 
+  #     E.g. valid & forecast_name are the same but primary_value is different
+  # For those instances, we are going to use the average forecasted value
+  group_by(forecast_name, valid) %>% 
+  summarize(primary_value = mean(primary_value)) %>% 
+  tidyr::spread(key = forecast_name, value = primary_value) %>% 
+  rename(dateTime = valid)
+
+##### Replace missing values with previously predicted value #####
```

Comment:
Fine for now, but we would probably want to be sure this is kosher for a finished product

Reply:
^ yes, agree. Seemed like carrying the old prediction for that timestep through if there wasn't any is what people who would use this data would do. But yeah 👍